src/coreclr/jit/hwintrinsicarm64.cpp

```diff
@@ -308,8 +308,19 @@ GenTree* Compiler::impSpecialIntrinsic(NamedIntrinsic        intrinsic,
                                        var_types             retType,
                                        unsigned              simdSize)
 {
-    HWIntrinsicCategory category = HWIntrinsicInfo::lookupCategory(intrinsic);
-    int                 numArgs  = sig->numArgs;
+    const HWIntrinsicCategory category = HWIntrinsicInfo::lookupCategory(intrinsic);
+    const int                 numArgs  = sig->numArgs;
+
+    // The vast majority of "special" intrinsics are Vector64/Vector128 methods.
+    // The only exception is ArmBase.Yield which should be treated differently.
+    if (intrinsic == NI_ArmBase_Yield)
+    {
+        assert(sig->numArgs == 0);
+        assert(JITtype2varType(sig->retType) == TYP_VOID);
+        assert(simdSize == 0);
+
+        return gtNewScalarHWIntrinsicNode(TYP_VOID, intrinsic);
+    }
 
```

Comment:
Nit: Add the following to catch things if we have more special cased scalar intrinsics.
```suggestion
assert((category != HW_Category_Scalar) && HWIntrinsicInfo::isScalarIsa(HWIntrinsicInfo::lookupIsa(intrinsic)));
```

Reply:
Done.